### ISSUE TITLE
Make the `certificate_authority` configuration item independent of the Cassandra configuration

### ISSUE THREAD

**Issue by jmdacruz** [Mashape/kong#1188]
Tl;DR: This is a recommendation on the current semantics for the `kong.yml` configuration file. There is a way to circumvent this limitation, but it seems that a minor change on the interpretation of the configuration file would be better: Having a global SSL client configuration to store things such as certificates to validate HTTPS calls to external services.

Currently, there is a configuration section that says something like this:

```yaml
  #####
  ## Client-to-node TLS options.
  ## `enabled`: if true, will connect to your Cassandra instance using TLS.
  ## `verify`: if true, will verify the server certificate using the given CA file.
  ## `certificate_authority`: an absolute path to the trusted CA certificate in PEM format used to verify the server certificate.
  ## For additional SSL settings, see the ngx_lua `lua_ssl_*` directives.
  # ssl:
  #   enabled: false
  #   verify: false
  #   certificate_authority: "/path/to/cluster-ca-certificate.pem"
```

This directive probably changes several things on the Cassandra client (based on the properties inside the `ssl` block), and it replaces the string `{{lua_ssl_trusted_certificate}}` on the nginx configuration with the value of `certificate_authority` on the YAML block above.

Now, this would be perfectly fine if the certificates on the nginx `lua_ssl_trusted_certificate` directive where only used for the Cassandra Lua client to talk to Cassandra, but in fact these certificates can also be used by a plugin to validate an HTTPS connection to an external service.

In this case, there are a couple of alternatives:
* Completely override de nginx directive and replace `{{lua_ssl_trusted_certificate}}` with the certificates you need (***this is what I'm doing right now on my plugin***). There are a few disadvantages with this approach:
   * If you want to enable SSL for Cassandra, you have to place your certificates manually on this configuration, since Kong won't be able to replace the `{{lua_ssl_trusted_certificate}}` directive anymore. Kong currently replaces this value with the nginx directive `lua_ssl_trusted_certificate` and the certificates themselves, which makes a chained replacement approach not possible (i.e., leaving the `{{lua_ssl_trusted_certificate}}` placeholder is not an option)
* Trying to reuse the `certificate_authority` property on the Cassandra section above, without setting `enabled: true`. The disadvantage in this case is that we don't know what the behavior is: One may assume that the `certificate_authority` property is only used if `enabled: true`, in which case this won't work. Even if Kong currently replaces the `{{lua_ssl_trusted_certificate}}` directive regardless on whether `enabled: true` or `enabled: false`, since this configuration is tied to Cassandra (at least that's what the documentation seems to imply) this behavior could change without notice.

So, the recommendation would be to have `certificate_authority` be a global property, independent on whether ssl is enabled or not for Cassandra. Something like this:

```YAML
  ## Global SSL trust configuration. For additional SSL settings, see the ngx_lua `lua_ssl_*` directives.
  # ssl:
  #   certificate_authority: "/path/to/cluster-ca-certificate.pem"

  ## Cassandra SSL trust configuration. Takes certificates from the global SSL configuration directives.
  # cassandra_ssl:
  #   enabled: false
  #   verify: false
```

**Comment by thibaultCha** [comment]
Yes, this is something we are aware of and we have plans to change its behavior.

**Comment by jmdacruz** [comment]
Great! And I should have mentioned we had a good conversation with @thibaultCha in Gitter around this topic (see https://gitter.im/Mashape/kong?at=5719476eb129b59c56da6007)

**Comment by jmdacruz** [comment]
@thibaultcha it seems that we should close this one too, right? Based on this https://github.com/Mashape/kong/blob/master/kong.conf.default#L257, it seems that this is already fixed in 0.9.x

**Comment by thibaultcha** [comment]
Yep! Forgot where to find this one, thanks.
